src/PhpSpec/Console/Provider/NamespacesAutocompleteProvider.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+/*
+ * This file is part of PhpSpec, A php toolset to drive emergent
+ * design by specification.
+ *
+ * (c) Marcello Duarte <marcello.duarte@gmail.com>
+ * (c) Konstantin Kudryashov <ever.zet@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace PhpSpec\Console\Provider;
+
+use Symfony\Component\Finder\Finder;
+
+final class NamespacesAutocompleteProvider
```

Comment:
There is attempt to create namespaces provider interface in #1078
If it makes sense to merge autoloading namespaces provider and autocomplete namespaces provider then you can implement it and name it in a way telling that it collects list of namespaces for autocompletion.